manifests/agent.pp

```diff
@@ -1,26 +1,45 @@
 class clamps::agent (
-  $nonroot_users = '2',
-  $master = $::servername,
-  $amqserver = $::servername,
-  $ca     = $::settings::ca_server,
-  $amqpass = file('/etc/puppetlabs/mcollective/credentials'),
+  $amqpass             = file('/etc/puppetlabs/mcollective/credentials'),
+  $amqserver           = [$::servername],
+  $ca                  = $::settings::ca_server,
+  $master              = $::servername,
+  $metrics_port        = 2003,
+  $metrics_server      = undef,
+  $nonroot_users       = '2',
+  $num_facts_per_agent = 500,
+  $shuffle_amq_servers = true,
 ) {
 
+  file { '/etc/puppetlabs/clamps':
+    ensure => directory
+  }
+
+  file { '/etc/puppetlabs/clamps/num_facts':
```

Comment:
The fact defaults to 500 if there is no file https://github.com/puppetlabs/clamps/pull/2/files#diff-501269a2e5a82c6ad9f759a9e593f562R12